client/cmd/dexc-desktop/app_darwin.go

```diff
@@ -96,41 +97,56 @@ func mainCore() error {
 		return fmt.Errorf("configuration error: %w", err)
 	}
 
+	// Return early if unsupported flags are provided.
+	if cfg.Webview != "" {
+		return errors.New("--webview flag is not supported. Other OS use it for a specific reason (to support multiple windows)")
+	}
+
+	// The --kill flag is a backup measure to end a background process (that
+	// presumably has active orders).
+	if cfg.Kill {
+		// This is a desktop app and the interactive quit or force quit
+		// button can be used to shutdown the app.
+		return errors.New(`--kill flag is not supported. Use the "Quit" or "Force Quit" button to kill any running process.`)
+	}
+
 	// Filter registered assets.
 	asset.SetNetwork(cfg.Net)
 
-	if cfg.Webview != "" { // Ignore, other OS use this for a specific reason (to support multiple windows).
-		return nil
+	// Use a "dexc-desktop-state" file to prevent other processes when
+	// dexc-desktop is already running (e.g when non-bundled version of
+	// dexc-desktop is executed from cmd and vice versa).
+	dexcDesktopStateFile := filepath.Join(cfg.Config.AppData, cfg.Net.String(), "dexc-desktop-state")
+	if dex.FileExists(dexcDesktopStateFile) {
+		return fmt.Errorf("dexc-desktop is already running, if not, manually delete this file: %s", strings.ReplaceAll(dexcDesktopStateFile, " ", `\ `))
+	}
+
+	err = os.WriteFile(dexcDesktopStateFile, []byte("running"), 0600)
```

Comment:
A related approach I have seen is to write the process ID to the file, so that when another instance starts, it can validate the lock file and delete it automatically if the process is already gone.
For example: https://github.com/go4org/go4/blob/214862532bf518db360e7ecc2b8a94b66a10c176/lock/lock.go#L83-L87

We wouldn't import that repo, but we could grab/vendor that `lock` package if it seems reasonable.

Reply:
Thanks for bringing this up. I will implement something related in my next commit after https://github.com/decred/dcrdex/pull/2372/commits/4682294509b267dfd81ae247deae6b7ceb42d3a6